### PR TITLE
Hud remake

### PR DESCRIPTION
**Pull request by TonybynMp4**
## Description

Remade the whole resource #5,
This probably needs some fancy compat layer.
### Features

- "Compass" 'inspired' by something @AnishBplayz did
   - Direction
   - Street
   - Zone
- Vehicle hud:
   - Speedometer
   - Fuel gauge
   - Blinkers
   - Headlights
   - Seatbelt / harness (same icon, different color)
   - Nitro gauge (using qbx_nitro)
- Player hud:
   - Health & Armor using the native ui
   - Thirst, Hunger & Stress (using statebags!!!)
   - Voice distance / activation, has a different color for talking over radio
- Money:
   - Bank & Cash
   - onMoneyChange event still works (kinda)
- Toggle hud / Cinematic mode
   - Toggleable with an event (to put in the radialmenu)
- Supports every stress feature of old hud

### Disclaimers & stuff
It's kinda like qbx_hud lean edition, it has less bloat than qb_hud, and is way simpler/better!
(Player hud subject to being hated on 😢)

The nitrous gauge icon is a placeholder, couldn't find a nice svg for it :c

To avoid errors due to "we're not currently using statebags for those you idiot",
search for every Functions.SetMetaData related to hunger or thirst and add
``
    Player(src).state:set('hunger', newHunger, true)
``
or
``
    Player(src).state:set('thirst', newThirst, true)
``
(modify it to work with the situation ofc)

Also the onMoneyChange stuff needs a rework, it's really not great

image

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

**Comment by TonybynMp4** [comment]
Someone please make a compat layer for hunger & thirst 😓

**Comment by jonezkti** [comment]
Any preview?

**Comment by TonybynMp4** [comment]
> Any preview?
:)

image
image
image

**Comment by AnishBplayz** [comment]
image
Try Removing this sleep variable... it should resolve the lint issue

**Comment by TonybynMp4** [comment]
fixed :)

**Comment by bicatcho11** [comment]
The old form is more beautiful and better at adding other interactions
I'm not into shape modification

**Comment by TonybynMp4** [comment]
wth are you yapping about

**Comment by TonybynMp4** [comment]
image
CUrrent version

File: client/player.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if value == PlayerState.hunger then return end
+    SendNUIMessage({
+        update = true,
+        data = {
+            {
+                type = 'progress',
+                name = 'hunger',
+                value = value,
+                option = {
+                    stroke = value < 30 and '#881111ff' or false,
```
Comment (BerkieBb) [inline client/player.lua:11]: The `or false` is not necessary as it turns false without it already if the conditions are not met

File: client/vehicle.lua [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+    CreateThread(function()
+        local sleep
+        while cache.seat == -1 do
+            sleep = 1000
+                if getIsVehicleEngineRunning(cache.vehicle) and isMinimapRendering() then
+                local HasTrailer, Trailer = getVehicleTrailerVehicle(cache.vehicle)
+                if isControlJustPressed(1, 174) then     -- <- is pressed
+                    indl = warning or not indl
+                    indr = indr and false
```
Comment (BerkieBb) [inline client/vehicle.lua:28]: Use the warning variable here as well to make sure it just doesn't turn off the indicator when those lights are on, if the warning variable is false, make indr just false, no need for that extra check

File: client/vehicle.lua [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                        type = 'dashboardlights',
+                        highbeam = highbeam,
+                        lowbeam = lowbeam,
+                    }
+                }
+
+
+                if config.lowFuelAlert and getVehicleFuelLevel(cache.vehicle) < config.lowFuelAlert then
+                    if alert > 0 then
+                        alert = alert - 1
```
Comment (BerkieBb) [inline client/vehicle.lua:114]: Use -=

File: client/vehicle.lua [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+        data = {
+            {
+                type = 'gauge',
+                name = 'nitro',
+                value = nitroLevel,
+                show = nitroLevel > 0
+            },
+            {
+                type = 'speedmax',
+                speed = GetVehicleEstimatedMaxSpeed(cache.vehicle) * 5.6 -- meh idk
```
Comment (BerkieBb) [inline client/vehicle.lua:158]: why the meh idk

File: client/voice.lua [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+---@param talking boolean
+local function setTalking(talking)
+    SendNUIMessage({
+        update = true,
+        data = {
+            {
+                type = 'progress',
+                name = 'voice',
+                option = {
+                    stroke = (PlayerState.radioActive and '#5A93FF') or (talking and '#FF935A') or false
```
Comment (BerkieBb) [inline client/voice.lua:14]: No need for the `or false`

**Review by BerkieBb — CHANGES_REQUESTED** [review]
Why do you just delete some locales?

**Comment by TonybynMp4** [comment]
> Why do you just delete some locales?

Don't want to keep locales that i cannot verify to be valid, especially since i changed things so old locales may not be good anymore

File: client/vehicle.lua (hunk 4 above)
Comment (TonybynMp4) [reply client/vehicle.lua:158]: It's a kinda random value, just noticed i should check for the mph/kmh too, but yeah it's just trying to be a bit above the estimated max speed, which can and mostly isn't the actual max speed

**Comment by ArturMichalak** [comment]
great job, the code looks much more developer friendly 👍

**Comment by ArturMichalak** [comment]
there is no information about the damage to the vehicle
obraz

**Comment by TonybynMp4** [comment]
> there is no information about the damage to the vehicle obraz

idk if it's really necessary, if the engine is about to die you'll know anyway 🤔

**Comment by ArturMichalak** [comment]
I don't have a problem with it if it was intentional

**Comment by TonybynMp4** [comment]
> I don't have a problem with it if it was intentional

👍🏻 Might consider adding an engine light or something like that tho 🤔

**Comment by ArturMichalak** [comment]
or engine Braking noises

**Comment by ArturMichalak** [comment]
I found this in smallresources.  It might be worth integrating this into qbx_hud

**Pull request closed without merging**